src/core/grid_based_algorithms/electrokinetics_cuda.cu

```diff
@@ -4045,10 +4045,9 @@ int ek_neutralize_system(int species) {
   return 0;
 }
 
-int ek_save_checkpoint(char *filename) {
+int ek_save_checkpoint(char *filename, char *lb_filename) {
   std::string fname(filename);
-  std::ofstream fout((const char *)(fname + ".ek").c_str(),
-                     std::ofstream::binary);
+  std::ofstream fout((const char *)(fname).c_str(), std::ofstream::binary);
```

Comment:
This look odd to me, why is this cast needed?

Reply:
it is not needed and I removed it